kubernetes/resource_kubernetes_service_test.go

```diff
@@ -811,6 +812,24 @@ func testAccCheckServicePorts(svc *api.Service, expected []api.ServicePort) reso
 	}
 }
 
+func testAccCheckloadBalancerIngressCheck(svc *api.LoadBalancerIngress) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		// Used with EKS Cluster
+		print(svc.Hostname)
+		print(svc.IP)
+		if svc.Hostname != "" {
```

Comment:
I'm a bit weary about relying just on the presence of this field value when trying to tell apart which cluster type we are running against.
We could try looking at the Node API objects instead and look for GKE specific labeling for example, or some other specific marker.

Reply:
Followed your advice and went with using `isRunningInGke()` helper function that checks for the `"cloud.google.com/gke-nodepool"` label within the cluster to verify that we're in GKE. The same can be found when determining if we're in a EKS cluster. Helps in checking for either `hostname` or `ip` depending on type of cluster.